src/cosmosShell/CosmosShellExtension.ts

```diff
@@ -0,0 +1,123 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * entry-point for mongoClusters-related code. Activated from ./src/extension.ts
+ *
+ * We'll try to have everything related to mongoClusters-support managed from here.
+ * In case of a failure with this plan, this comment section will be updated.
+ */
+import {
+    callWithTelemetryAndErrorHandling,
+    registerCommandWithTreeNodeUnwrapping,
+    type IActionContext,
+} from '@microsoft/vscode-azext-utils';
+import * as l10n from '@vscode/l10n';
+import * as child from 'child_process';
+import * as vscode from 'vscode';
+import { ext } from '../extensionVariables';
+import { type NoSqlContainerResourceItem } from '../tree/nosql/NoSqlContainerResourceItem';
+
+export class CosmosShellExtension implements vscode.Disposable {
+    dispose(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    async activate(): Promise<void> {
+        await callWithTelemetryAndErrorHandling('cosmosDB.cosmosShell.activate', (_activateContext: IActionContext) => {
+            const isCosmosShellInstalled: boolean = isCosmosShellSupportEnabled();
+            vscode.commands.executeCommand(
+                'setContext',
+                'vscodeDatabases.cosmosShellSupportEnabled',
+                isCosmosShellInstalled,
+            );
+            registerCommandWithTreeNodeUnwrapping('cosmosDB.launchCosmosShell', launchCosmosShell);
+            if (isCosmosShellInstalled) {
+                ext.outputChannel.appendLine(`Cosmos Shell Extension: activated.`);
+            } else {
+                ext.outputChannel.appendLine(`Cosmos Shell Extension: deactivated.`);
+            }
+        });
+    }
+}
+
+function getCosmosShellCommand(): string {
+    const config = vscode.workspace.getConfiguration();
+    const shellPath: string | undefined = config.get('cosmosDB.shell.path');
+    return shellPath || 'CosmosShell';
+}
+
+export function launchCosmosShell(_context: IActionContext, node?: NoSqlContainerResourceItem) {
+    const isCosmosShellInstalled: boolean = isCosmosShellSupportEnabled();
+
+    if (!isCosmosShellInstalled) {
+        void vscode.window.showErrorMessage(
```

Comment:
Let's add a "Settings" button which opens the settings URL: "vscode://settings/cosmosDB.shell.path"

Reply:
fixed.